### PR TITLE
Transform

### PR DESCRIPTION
**Pull request by rafaelfolco**

### CODE REVIEW

**Comment by rafaelfolco** [comment]
under testing yet, not ready for review

**Comment by rafaelfolco** [comment]
unit tests to be added after we add schema validation for the requirements file.... for now only tested locally with some test cases using JSON/*trafficgen.json samples.

File: JSON/requirements-sample-fio.json [hunk 1]
```diff
@@ -14,12 +14,15 @@
...
-        "size_KMG": {
-            "description": "bytes in k/K (1024), m/M (1024^2) or g/G (1024^3): 4k 16M 1g",
+        "size_BKMG": {
+            "description": "bytes in B/K/M/G, converted to K, shown in KB",
             "args": [ "bs" ],
-            "vals": "[0-9]+[kbmgKBMG]",
-            "transform": [ "s/([0-9]+)[gG]/($1*1024).\"M\"/e",
-                           "s/([0-9]+)[mM]/($1*1024).\"K\"/e" ]
+            "vals": "[0-9]+[BKMG]?",
+            "convert": "",
```
Comment (k-rister) [inline JSON/requirements-sample-fio.json:21]: I thought this would have to be `"convert": "KB"`?

File: JSON/requirements-sample-fio.json [hunk 2]
```diff
@@ -14,12 +14,15 @@
...
+            "transform": {
+                "search": "[^0123456789\\.]",
```
Comment (k-rister) [inline JSON/requirements-sample-fio.json:23]: Are you sure you don't mean `"search": "[^[1-9][0-9]*\\.]"`?  As written I think this is exclusively looking for the string `0123456789`.

File: JSON/requirements-sample-trafficgen.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    "presets": {
+    },
+    "validations": {
+        "size_BKMG": {
+            "description": "bytes in B/K/M/G, converted to K, shown in KB",
+            "args": [ "frame-size" ],
+            "vals": "[0-9]+[BKMG]?",
+            "convert": "K",
+            "transform": {
+                "search": "[^0123456789\\.]",
```
Comment (k-rister) [inline JSON/requirements-sample-trafficgen.json:11]: Same as above.

File: README.md [hunk 4]
```diff
@@ -92,12 +92,15 @@ version of fio benchmark requirements file:
...
-            "vals" : "[0-9]+[kbmgKBMG]",
-            "transform" : [ "s/([0-9]+)[gG]/($1*1024).\"M\"/e",
-                            "s/([0-9]+)[mM]/($1*1024).\"K\"/e" ]
+        "size_BKMG": {
+            "description": "bytes in B/K/M/G, converted to K, transformed in KB",
+            "args": [ "frame-size" ],
+            "vals": "[0-9]+[BKMG]?",
+            "convert": "K",
+            "transform": {
+                "search": "[^0123456789\\.]",
```
Comment (k-rister) [inline README.md:101]: again

File: README.md [hunk 5]
```diff
@@ -141,11 +153,33 @@ need to function properly. Theses parameters are always appended to the list of
...
 `value` regex. The values are transformed by applying the `transform` regex,
-if present. The `transform` key is optional.
+if present. The `transform` key (optional) contains the `search` and `replace`
+regular expressions to substitute the `vals` key from the multi-value params
+file.
+
+Note: The regex pair `search`/`replace` must be in raw string format to
+match and process the substitutions w/ backslashes (escapes) properly. Also,
+since the JSON has its own escaping, backslashes must be doubled. For instance,
+"[^0123456789\.]" becomes "[^0123456789\\.]" to match all numeric values w/
```
Comment (k-rister) [inline README.md:166]: same

File: JSON/requirements-sample-fio.json (hunk 2 above)
Comment (rafaelfolco) [reply JSON/requirements-sample-fio.json:23]: It doesn't matter, this one is not to validate leading zeros. I can change it to make it consistent, though.

File: JSON/requirements-sample-trafficgen.json (hunk 3 above)
Comment (rafaelfolco) [reply JSON/requirements-sample-trafficgen.json:11]: ack

File: README.md (hunk 4 above)
Comment (rafaelfolco) [reply README.md:101]: ack

File: README.md (hunk 5 above)
Comment (rafaelfolco) [reply README.md:166]: ack

**Review by rafaelfolco — COMMENTED** [review]
thx for reviewing, updating the PR now...

File: JSON/requirements-sample-fio.json (hunk 1 above)
Comment (rafaelfolco) [reply JSON/requirements-sample-fio.json:21]: yes, it must be "K", sorry.

**Review by k-rister — APPROVED** [review]

**Pull request merged**